fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/JobsServlet.java

```diff
@@ -431,7 +454,44 @@ protected void handleResultsResponse(
             );
             return Observable.error(responseException);
         }
-        return Observable.just(preResponse);
+
+        return paginationParameters
+                .map(pageParams -> new AllPagesPagination<>(preResponse.getResultSet(), pageParams))
+                .map(pages -> new PreResponse(
```

Comment:
Nitpick: Should probably call the lambda argument `page` rather than `pages` since it only contains one page of data.